### ISSUE TITLE
Multiple anonymous System.register calls in the same module file.

### ISSUE THREAD

**Issue by jamiewinder** [Microsoft/TypeScript#3937]
I've taken a project I was working on from the 1.5 beta, and simply switched from CommonJS modules to System. I now seem to be getting the title error on startup which suggests something is perhaps amiss in the output?

I've not managed to create a small reproduction yet, but here are the TS and JS for the code file which SystemJS is complaining about:

alert.ts
```typescript
import { StandardDialog } from '../standard/standard';

export interface AlertDialogOptions {
    /**
     * The text to display on the 'OK' button. Defaults to 'OK'.
     */
    okText?: string;
}

export class AlertDialog extends StandardDialog {
    // Constructor
    constructor(text: string, opts?: AlertDialogOptions) {
        const content = $("<p>").text(text);        
        const okText = (opts && opts.okText) ? opts.okText : "OK";

        super({
            content: content,
            buttons: [
                { text: okText, primary: true, click: () => this._ok_click() }
            ]
        });

        this.el.classList.add("alert-dialog");
    }

    // Event Handlers
    private _ok_click(): void {
        this.close();
    }

    // Static Methods

    /**
     * Shows an Alert dialog. Returns a promise that will resolve when the OK button is clicked
     * @param text the text to be displayed
     * @param opts any non-default options
     */
    public static show(text: string, opts?: AlertDialogOptions): Promise<void> {
        const dlg = new AlertDialog(text, opts);
        return dlg.show();
    }
}
```

alert.js
```javascript
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    __.prototype = b.prototype;
    d.prototype = new __();
};System.register(['../standard/standard'], function(exports_1) {
    var standard_1;
    var AlertDialog;
    return {
        setters:[
            function (_standard_1) {
                standard_1 = _standard_1;
            }],
        execute: function() {
            AlertDialog = (function (_super) {
                __extends(AlertDialog, _super);
                // Constructor
                function AlertDialog(text, opts) {
                    var _this = this;
                    var content = $("<p>").text(text);
                    var okText = (opts && opts.okText) ? opts.okText : "OK";
                    _super.call(this, {
                        content: content,
                        buttons: [
                            { text: okText, primary: true, click: function () { return _this._ok_click(); } }
                        ]
                    });
                    this.el.classList.add("alert-dialog");
                }
                // Event Handlers
                AlertDialog.prototype._ok_click = function () {
                    this.close();
                };
                // Static Methods
                /**
                 * Shows an Alert dialog. Returns a promise that will resolve when the OK button is clicked
                 * @param text the text to be displayed
                 * @param opts any non-default options
                 */
                AlertDialog.show = function (text, opts) {
                    var dlg = new AlertDialog(text, opts);
                    return dlg.show();
                };
                return AlertDialog;
            })(standard_1.StandardDialog);
            exports_1("AlertDialog", AlertDialog);
        }
    }
});
//# sourceMappingURL=alert.js.map
```
Console:
```
Uncaught TypeError: Multiple anonymous System.register calls in the same module file.
Uncaught (in promise) Uncaught TypeError: Multiple anonymous System.register calls in the same module file.
	Evaluating http://localhost:4544/src/ui/dialogs/common/alert.js
	Error loading http://localhost:4544/src/startup.js
```
Hope this is helpful. I'll try to find a way to reproduce with a smaller set of code if it isn't.

EDIT: btw, I did some searching around the issue before posting and noticed a few things said about bundles. Just to be clear, I'm not using bundles at the moment.

**Comment by jamiewinder** [comment]
After some more digging around, I think this is a known issue with files that emit the __extends function - it stops SystemJS detecting the module format. I got around it for now by adding the "format register" string to the top of all affected files.

**Comment by mhegazy** [comment]
duplicate of #3655

**Comment by jamiewinder** [comment]
Is it a duplicate? #3655 seems to be talking about the naming of modules. The issue here seems to be that TypeScript emits the __extends function before the System.register call which is preventing SystemJS from correctly detecting the module format.

**Comment by dsebastien** [comment]
I have the same issue since I've tried to switch to the "system" module format.

Basically as soon as I switch, my app (Angular2 w/ SystemJS & JSPM) stops loading and I get no error so it's not very straightforward to debug.

By adding `'format register';` at the top of all ts files the issue seems to be resolved but I suppose that it only helps SystemJS understand/load the modules correctly..

**Comment by mhegazy** [comment]
@jamiewinder sorry i did not explain. your diagnosis is correct. @yui is fixing it by ensuring that the `__extends` is emitted inside the body.

**Comment by mhegazy** [comment]
another workaround is to use `--noEmitHelpers` to suppress emitting the `__extends` in all files, but you need to include it at least once in your project.

**Comment by dashaus** [comment]
I have the same problem with my project. Currently, I switched to "umd" and this works but I would like come back to use SystemJS!
